cardano-chain-gen/test/Test/Cardano/Db/Mock/Unit/Babbage/Config/MigrateConsumedPruneTxOut.hs

```diff
@@ -54,338 +53,333 @@ import Test.Cardano.Db.Mock.UnifiedApi (
 import Test.Cardano.Db.Mock.Validate (assertBlockNoBackoff, assertEqQuery, assertTxCount, assertTxInCount, assertTxOutCount, assertUnspentTx, checkStillRuns)
 import Test.Tasty.HUnit (Assertion)
 
+migrateTxOutPruneForceTxIn :: SyncNodeConfig -> SyncNodeConfig
+migrateTxOutPruneForceTxIn cfg = do
+  cfg {dncInsertOptions = (dncInsertOptions cfg) {sioTxOut = TxOutPrune (ForceTxIn True)}}
+
+migrateTxOutPruneConfig :: SyncNodeConfig -> SyncNodeConfig
+migrateTxOutPruneConfig cfg = do
+  cfg {dncInsertOptions = (dncInsertOptions cfg) {sioTxOut = TxOutPrune (ForceTxIn False)}}
+
+migrateTxOutConsumeConfig :: SyncNodeConfig -> SyncNodeConfig
+migrateTxOutConsumeConfig cfg = do
+  cfg {dncInsertOptions = (dncInsertOptions cfg) {sioTxOut = TxOutConsumed (ForceTxIn False)}}
+
 txConsumedColumnCheck :: IOManager -> [(Text, Text)] -> Assertion
-txConsumedColumnCheck = do
-  withCustomConfigAndDropDB cmdLineArgs Nothing babbageConfigDir testLabel $ \interpreter mockServer dbSyncEnv -> do
-    void $
-      withBabbageFindLeaderAndSubmitTx interpreter mockServer $
-        Babbage.mkPaymentTx (UTxOIndex 0) (UTxOIndex 1) 10000 500
-
-    startDBSync dbSyncEnv
-    assertBlockNoBackoff dbSyncEnv 1
-    assertEqQuery dbSyncEnv DB.queryTxConsumedColumnExists True "missing consumed_by_tx_id column when flag --consumed-tx-out active"
+txConsumedColumnCheck ioManager metadata = do
+  syncNodeConfig <- mkCustomSyncNodeConfig babbageConfigDir args migrateTxOutPruneForceTxIn
+  withCustomConfig args (Just syncNodeConfig) babbageConfigDir testLabel action ioManager metadata
```

Comment:
Would it be possible to change the `withCustomConfig` definition, so that it takes the `migrateTxOutPruneForceTxIn` and similar as an argumant? That way the `args` and `babbageConfigDir` don't have to be used 2 times here and the format of tests don't need to change.

Reply:
I wanted to do that but `withCustomConfig` is used in a lot of places. But happy to do it as we might as most of these files are the same and it's just insert-options that differ

Reply:
My main issue with these is that the primary testing logic is buried in a `where` clause. What if wee moved `action` to the end, so it would just be a standard continuation?

Reply:
it looks a lot better when I put the logic inside of `withCustomConfig` then the testing logic isn't in a where clause :)